packages/app-settings/src/index.tsx

```diff
@@ -178,9 +210,40 @@ class App extends React.PureComponent<Props, State> {
     }));
   }
 
+  private toggleCustomNode = (): void => {
+    this.setState(({ settings }: State) => ({
+      settings: {
+        ...settings,
+        /*flip the value of customNode*/
+        customNode: settings.customNode ?
+        false
+        :
+        true
+      }
+    }));
+  }
+
   private save = (): void => {
+    const { onStatusChange } = this.props;
     const { settings: { types, typesError } } = this.state;
 
+    // validate custom node url
+    const apiUrl = this.state.settings.apiUrl;
+
+    if (this.state.settings.customNode) {
```

Comment:
* Are we transporting any MSRP traffic over WebSockets? If so, should we only allow WSS and have the client authenticate the server by validating the server's certificate? Even if the user is using WSS should we not indicate end-to-end security to the user as specified by RFC 7977 "9. Security Considerations" https://tools.ietf.org/html/rfc7977#page-24?
* Are there any risks associated with using a custom WSS or WS connection that we should communicate to the user?
* If they connect using WS then what are they exposing to a potential MITM attack?
* Are we exposing the user to [Cross-Site WebSocket Hijacking (CSWSH)](https://content-security-policy.com/)?
* Do we need to add a CSP like `<meta http-equiv="Content-Security-Policy" content="connect-src 'self'">` to prevent WS requests, CORS, EventSource requests from anywhere other than the current server?